agilent_infiniium_scope.py

```diff
@@ -31,446 +31,344 @@
-        if scale is None:
-            resp = yield dev.query('TIM:SCAL?')
-        else:
-            scale = format(scale,'E')
-            yield dev.write('TIM:SCAL '+scale)
-            resp = yield dev.query('TIM:SCAL?')
+        if scale is not None:
+            yield dev.write('TIM:SCAL {:E}'.format(scale))
+        resp = yield dev.query('TIM:SCAL?')
         scale = float(resp)
         returnValue(scale)
     
-    #Data acquisition settings
-    @setting(201, channel = 'i', start = 'i', stop = 'i', returns='*v[ns] {time axis} *v[mV] {scope trace}')
+    # Data acquisition settings
+    @setting(201, channel='i', start='i', stop='i',
+             returns='*v[ns] {time axis} *v[mV] {scope trace}')
     def get_trace(self, c, channel, start=1, stop=10000):
         """Get a trace from the scope.
+
         OUTPUT - (array voltage in volts, array time in seconds)
         """
-        #raise Exception('Doesnt work yet. Please fix lines defining traceVolts and time.')
-##        DATA ENCODINGS
-##        RIB - signed, MSB first
-##        RPB - unsigned, MSB first
-##        SRI - signed, LSB first
-##        SRP - unsigned, LSB first
-        wordLength = 2 #Hardcoding to set data transer word length to 2 bytes
+        # DATA ENCODINGS
+        # RIB - signed, MSB first
+        # RPB - unsigned, MSB first
+        # SRI - signed, LSB first
+        # SRP - unsigned, LSB first
+        word_length = 2  # Hardcoding to set data transer word length to 2 bytes
         
         dev = self.selectedDevice(c)
-        yield dev.write('WAV:SOUR CHAN%d' %channel)
+        yield dev.write('WAV:SOUR CHAN{}'.format(channel))
 
-        #Read data MSB first
+        # Read data MSB first
         yield dev.write('WAV:BYT MSBF')
-        #Set 2 bytes per point
+        # Set 2 bytes per point
         yield dev.write('WAV:FORM WORD')
-        #Starting and stopping point
-        #Transfer waveform preamble
+        # Starting and stopping point
+        # Transfer waveform preamble
         preamble = yield dev.query('WAV:PRE?')
-        #Transfer waveform data
-        binary = yield dev.query('WAV:DATA?')
-        #Parse waveform preamble
-        preambleDict = _parsePreamble(preamble)
-        print preambleDict
-        #Parse binary
-        trace = _parseBinaryData(binary, wordLength = wordLength) * 1.0e3
-        #Convert from binary to volts
-        traceVolts = ((trace*float(preambleDict['yStep'])+float(preambleDict['yOrigin'])))
-        numPoints = int(preambleDict['numPoints'])
-        time = numpy.linspace(float(preambleDict['xFirst']),
-                              (numPoints-1) * float(preambleDict['xStep'])+float(preambleDict['xFirst']),
-                              numPoints)
-
-        returnValue((time*U.ns*1e9, traceVolts*U.V))
+        # Transfer waveform data
+        p = dev._packet().write('WAV:DATA?').read_raw()
+        result = yield p.send()
+        binary = result['read_raw'][:-1]  # drop termination character
```

Comment:
I just realized that the new implementation of `_parseBinaryData` that I suggested expects this termination character to be there when it looks at the total data length. So no longer need to drop it here.